analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ExecutingSqlQueries.cs

```diff
@@ -81,10 +82,56 @@ public ExecutingSqlQueries()
             && expression is BinaryExpressionSyntax concatenation
             && !IsConcatenationOfConstants(concatenation, semanticModel);
 
-        private bool IsTrackedVariableDeclaration(ExpressionSyntax argument, SemanticModel semanticModel) =>
-            (argument is IdentifierNameSyntax identifierNameSyntax
-             && semanticModel.GetDeclaringSyntaxNode(identifierNameSyntax) is VariableDeclaratorSyntax variableDeclaratorSyntax
-             && IsTracked(variableDeclaratorSyntax.Initializer?.Value, semanticModel));
+        private static bool IsTrackedVariableDeclaration(ExpressionSyntax expression, BaseContext context)
+        {
+            if (expression is IdentifierNameSyntax)
+            {
+                var node = expression;
+                Location location;
+                do
+                {
+                    var identifierName = (node as IdentifierNameSyntax).Identifier.ValueText;
+                    node = AssignmentFinder.FindLinearPrecedingAssignmentExpression(identifierName, node) as ExpressionSyntax;
+                    location = GetSecondaryLocationForExpression(node);
+
+                    if (IsSensitiveExpression(node, context.SemanticModel))
+                    {
+                        if (location != Location.None)
+                        {
+                            context.AddSecondaryLocation(new SecondaryLocation(location, string.Format(AssignmentWithFormattingMessage, identifierName)));
+                        }
+
+                        return true;
+                    }
+                    else
+                    {
+                        if (location != Location.None)
```

Comment:
Would it make sense to move this condition inside `AddSecondaryLocation` instead of the `Debug.Assert`? To remove the responsibility from each user and keep the class responsible for it's own state? You wouldn't need the `location` variable then.